lib/compat/glib.h

```diff
@@ -53,6 +53,27 @@ GList *g_list_copy_deep (GList *list, GCopyFunc func, gpointer user_data);
     (void) (0 ? (val) ^ (val) : 0);                                          \
     (gssize) __sync_fetch_and_add ((atomic), (val));                         \
   }))
+#define g_atomic_pointer_or(atomic, val) \
+  (G_GNUC_EXTENSION ({                                                       \
+    G_STATIC_ASSERT (sizeof *(atomic) == sizeof (gpointer));                 \
+    (void) (0 ? (gpointer) *(atomic) : 0);                                   \
+    (void) (0 ? (val) ^ (val) : 0);                                          \
+    (gsize) __sync_fetch_and_or ((atomic), (val));                           \
+  }))
+#define g_atomic_pointer_xor(atomic, val) \
```

Comment:
All these defines are put `#ifndef g_atomic_pointer_add` guardian.
I miss the `#ifndef g_atomic_pointer_xor` etc. Is it safe to depend everything on `g_atomic_pointer_add`?

Reply:
At least they have been introduced in the same commit https://github.com/GNOME/glib/commit/83821352657a9481dbff6ab04e8ae60566c17d5e 
But if you question that, I think the check itself also questionable (see if the `g_atomic_pointer_add` has been defined as macro...)

Reply:
Being in the same commit is convincing enough for me. Thanks.